imports/client/components/CallLinkBox.tsx

```diff
@@ -274,22 +274,24 @@ class CallLinkBox extends React.PureComponent<CallLinkBoxProps, CallLinkBoxState
       >
         <div
           key={`viewer-${this.props.peerParticipant._id}`}
-          className="people-item"
+          className={`people-item ${this.props.peerParticipant.muted ? 'muted' : ''} ${this.props.peerParticipant.deafened ? 'deafened' : ''} ${this.props.peerParticipant.muted || this.props.peerParticipant.deafened ? '' : 'active'}`}
         >
           <span className="initial">{name.slice(0, 1)}</span>
           <div className="webrtc">
+            {this.props.peerParticipant.muted && <span className="icon muted-icon"><FontAwesomeIcon icon={faMicrophone} /></span>}
+            {this.props.peerParticipant.deafened && <span className="icon deafened-icon"><FontAwesomeIcon icon={faHeadphonesAlt} /></span>}
+            {!this.props.spectraDisabled &&
+              !this.props.peerParticipant.muted &&
```

Comment:
This doesn't reliably come back when the peer mutes and unmutes, because `Spectrum` needs to have `connect` called on it to provide it with a stream.  (To the extent that it does work by coincidence, it must be relying on object reuse implementation details that are far from guaranteed.)

The other Spectrum instance is fine because it uses a callback ref to ensure we propagate the stream that we already have laying around at the time of rendering; it's possible we could do so here if we have the track already but we'd still need to save a handle since we might not have the stream/track from the RTCPeerConnection yet.  Similarly, we'd need to modify the onRemoteTrack callback to do the same "check if we have both objects, connect if so" logic.

Reply:
I don't really understand this comment? The Spectrum component still has the ref passed to it, it's just further down the file. And the icons already refer to `peerParticipant.muted`...

Reply:
Sorry, let me try again.

Just putting a `<Spectrum>` in the component tree will not get you any visible spectra, because there's no media stream being fed into it yet.  We feed in that media stream in exactly one place in this file; in the `onNewRemoteTrack` callback.  So if the `Spectrum` element is not around when that single callback fires, or vanishes and returns, then the Spectrum element will have no audio feed to visualize.

Repro steps:

1. Have two callers join a call, call them Participant A and Participant B.
2. Participant B mutes or deafens.
3. Participant A sees it and removes the `<Spectrum>` from the render.
4. Participant B unmutes/undeafens.
5. Participant A sees it and adds `<Spectrum>` to the render again
6. But nothing ever calls `.connect` on the instance of Spectrum in step 5.
7. So Participant A sees only silence in the spectrum drawn for Participant B forevermore.

The reason this isn't *also* an issue in CallSection is because we always have the local microphone stream in advance of the Spectrum getting rendered, so we call `.connect` in the callback ref in CallSection at https://github.com/deathandmayhem/jolly-roger/pull/343/files#diff-a9128bbe54b44b926e42180c5397ea0a33dd7256546bf5fd524b355c7fde60b6R90

There's a couple options for fixing this:

1. Make Spectrum take a stream as a prop from its parent, and carefully wire up the analyserNode in componentDidMount/componentDidUpdate.  We'd need to hold on to the `stubStream` in `CallLinkBox`.
2. Alternately, we could make `CallLinkBox` track when both the `Spectrum` and the stream are ready, and call `spectrumRef.current.connect(theStream)` in each case where appropriate.  That's what I tried to propose in the previous comment.

Reply:
To add additional comment: this bug is fundamentally my fault and was always an issue if we toggled the `spectraDisabled` feature flag too!  I just figured that probably wouldn't come up in practice with the feature flag, since we probably don't toggle that a whole lot (and if we turn it off we probably leave it off).  Having this break on mute/unmute seems like substantially more exposure.

Worst case I can always take this patch and just take a crack at fixing the bug myself